.github/workflows/checks.yml

```
@@ -0,0 +1,35 @@
+name: Checks
+
+on:
+  pull_request:
+
+jobs:
+
+  lint-job:
+    name: Linting and Type checks (Python-${{ matrix.python-version }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.14.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
+        run: poetry run nox -s lint
+
+      - name: Run type-check
+        run: poetry run nox -s type-check
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
```

```suggestion
        uses: actions/upload-artifact@v4
```
You might need to use a unique artifact name if you still use multiple python versions.

---

Done